app/controllers/badges_controller.rb

```diff
@@ -16,7 +16,7 @@ def require
   end
 
   def new
-    if user_signed_in? && current_user.is_organizer
+    if user_signed_in? && current_user.is_organizer && current_user.is_badge_guardian
```

Comment:
You do know we can still do XSS to update badges, right? This should be done using `cancancan`.